Bottom: cbb8033e60f755ec64df59d5451fb4a55c15a8fc
Top:    b315cdb38da542396fa7392bb1909e1cb3689508
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 20:56:22 +0000

find.c: Fix inverted logic in LookupConnection()

When AndRats is true, we may consider starting with a ratline object,
otherwise we do no. I got this backwards by mistake.


---

diff --git a/src/find.c b/src/find.c
index dd38281..d6ec2ec 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3077,7 +3077,7 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
   if (type == NO_TYPE)
     {
       type = SearchObjectByLocation (
-        LOOKUP_MORE & ~(AndRats ? RATLINE_TYPE : 0),
+        LOOKUP_MORE & ~(AndRats ? 0 : RATLINE_TYPE),
         &ptr1, &ptr2, &ptr3, X, Y, Range);
       if (type == NO_TYPE)
         return;
